roles/snapshot_revert/tasks/main.yml

```
@@ -66,7 +66,9 @@
 - name: Wait for the snapshot to drain
   ansible.builtin.command: "lvs --select 'vg_name = {{ item.vg_name }} && lv_name = {{ item.origin }}' --reportformat json"
   register: snapshot_revert_lv_drain_check
-  until: (snapshot_revert_lv_drain_check.stdout | from_json).report[0].lv[0].data_percent == ""
+  until: >-
+    (snapshot_revert_lv_drain_check.stdout | from_json).report[0].lv[0].data_percent == "" or
+    (snapshot_revert_lv_drain_check.stdout | from_json).report[0].lv[0].lv_attr == "Vwi-aotz--"
```

I've been playing around with thick snapshots versus thin snapshots. One thing I learned is that the behavior is very different with the `lvconvert --merge` we do in the revert role. 

With thick snapshots, this idea of "draining" makes sense to describe the time it takes while the copy-on-write (COW) blocks are merged back to the origin upon the volume being reactivated after the reboot.

However, thin snapshots don't use COW. After `lvconvert --merge` of a thin snapshot and reboot, the content of the thin snapshot instantly becomes the content of the origin volume when it is reactivated and snapshot volume instantly disappears.

Since there is no "drain" to wait for in the case of thin snapshots, I think we should just move on if `lv_attr[0] == "V"`, that is, if it is thin...
```suggestion
    (snapshot_revert_lv_drain_check.stdout | from_json).report[0].lv[0].lv_attr[0] == "V"
```

---

Thanks @swapdisk  - great input!

Have updated the code with these suggestions - am currently testing them.

---

Cool, I will test as well.